packages/edit-post/src/components/layout/index.js

```diff
@@ -136,6 +129,27 @@ function Layout() {
 		}
 	}, [ isInserterOpen, isHugeViewport ] );
 
+	// Local state for save panel.
+	const [
+		isEntitiesSavedStatesOpen,
+		setIsEntitiesSavedStatesOpen,
+	] = useState( false );
+	const [ closeEntitiesCallback, setCloseEntitiesCallback ] = useState();
+	const openEntitiesSavedStates = useCallback(
+		() => setIsEntitiesSavedStatesOpen( true ),
+		[]
+	);
+	const closeEntitiesSavedStates = useCallback(
+		( callbackArg ) => {
+			if ( closeEntitiesCallback ) {
+				closeEntitiesCallback( callbackArg );
+				setCloseEntitiesCallback();
+			}
+			setIsEntitiesSavedStatesOpen( false );
+		},
+		[ closeEntitiesCallback ]
+	);
+
```

Comment:
```js
const [
	entitiesSavedStatesCallback,
	setEntitiesSavedStatesCallback,
] = useState( false );
const closeEntitiesSavedStates = useCallback(
	( arg ) => {
		if ( typeof entitiesSavedStatesCallback === 'function' ) {
			entitiesSavedStatesCallback( arg );
		}
		setEntitiesSavedStatesCallback( false );
	},
	[ entitiesSavedStatesCallback ]
);
```

Non-blocking, but it would be simpler.

```js
if ( entitiesSavedStatesCallback ) {
	// It's open.
}

// Open with callback.
setEntitiesSavedStatesCallback( () => () => {} );

// Just open.
setEntitiesSavedStatesCallback( true );
```

Reply:
That is really neat!  Thank you for the suggestion, I will apply this as it does seem to clean up a couple things.